### DISCUSSION TITLE
You are rate-limited Rate-limited or API key is invalid

### DISCUSSION THREAD

**Discussion by JerryWn12** [waylaidwanderer/node-chatgpt-api#60] [Help] [answered]
using reverse proxy
proxy: https://chatgpt.pawan.krd/api/completions
model: text-davinci-002-render
api_key: https://chat.openai.com/api/auth/session's accessToken

**Comment by JerryWn12** [comment]
fixed

**Comment by PwLDev** [comment]
How did you fixed it? I'm having the same problem

**Comment by waylaidwanderer** [comment]
I heard it might be a bug with Pawan's API server. Use a free account with the `text-davinci-002-render` model.

**Comment by kilingzhang** [comment]
+1

**Comment by AutumnWhj** [comment]
How did you fixed it? I'm having the same problem

**Comment by JerryWn12** [comment]
I didn't do anything it just work automaticly🥲sorry just see the message
@PwLDev @kilingzhang @AutumnWhj

**Comment by AutumnWhj** [comment]
can you share your config ? 👁

**Comment by JerryWn12** [comment]
```
module.exports = {
    openaiApiKey: process.env.OPENAI_API_KEY || 'access_token_got_from_https://chat.openai.com/api/auth/session',
    chatGptClient: {
        reverseProxyUrl: 'https://chatgpt.pawan.krd/api/completions',
        modelOptions: {
            model: 'text-chat-davinci-002',
        },
        debug: false,
    },
    cacheOptions: {},
    bingAiClient: {
        userToken: '',
        debug: false,
    },
    apiOptions: {
        port: process.env.API_PORT || 3000,
        host: process.env.API_HOST || 'localhost',
        debug: false,
    },
    storageFilePath: process.env.STORAGE_FILE_PATH || './cache.json',
};
```
here is my config @AutumnWhj

**Comment by waylaidwanderer** [comment]
I see you set the `model` to `text-chat-davinci-002`. I think Pawan's API sets it to the `text-davinci-002-render` model if you don't give it a correct model name (like in your case), so keep that in mind.

**Comment by JerryWn12** [comment]
Thanks i will check my available model

**Comment by kilingzhang** [comment]
Still no effect  ಥ_ಥ  @AutumnWhj Does it work?

**Comment by waylaidwanderer** [comment]
Trying @acheong08's API server instead: https://chat.duti.tech/completions

Let me know if it works for you guys.

Edit: currently, his API endpoint throws an error due to the model parameter being set (his server doesn't accept extra parameters). He'll be updating it to allow setting the model, but this won't work until then.

**Reply by kilingzhang** [reply]
`{"message":"Invalid request"}`

**Comment by waylaidwanderer** [comment]
I'll try to deploy my own API server tomorrow as well btw

**Comment by waylaidwanderer** [comment, marked as answer]
For now, if you're having this issue, make sure to use a non-Plus account.

**Reply by kilingzhang** [reply]
ummm.  Ok, I'll change to a non-plus account

**Reply by kilingzhang** [reply]
Still no effect ಥ_ಥ. It seems that I can only wait for the server deployed by you tomorrow to check again.

**Comment by JerryWn12** [comment]
Hey guys please notice that if you are using the reverse proxy, the `openaiApiKey` in settings.js should be the accessToken copied from [https://chat.openai.com/api/auth/session](https://chat.openai.com/api/auth/session) but **NOT** the [https://platform.openai.com/account/api-keys](https://platform.openai.com/account/api-keys)
@PwLDev @kilingzhang @AutumnWhj

**Reply by JerryWn12** [reply]
<del>it seems like they patched this, the website no longer return the accessToken</del>
\*my mistake, i haven't login to [chat.openai.com](https://chat.openai.com/chat), after that the website returns accessToken again

**Reply by kilingzhang** [reply]
yes i confirm access_token not apikey ಥ_ಥ
